ui/jobs/utils.ts

```diff
@@ -1,76 +1,43 @@
-import { LocaleNetRegex } from '../../resources/translations';
+import { getLocaleRegexes } from '../../resources/translations';
 import Util from '../../resources/util';
 import { Job } from '../../types/job';
-import { CactbotBaseRegExp } from '../../types/net_trigger';
+import { NetAnyFields } from '../../types/net_fields';
 
 import { Bars } from './bar';
 import { kLevelMod, kMeleeWithMpJobs } from './constants';
 
-const getLocaleRegex = (locale: string, regexes: {
-  'en': RegExp;
-  [x: string]: RegExp;
-}): RegExp => regexes[locale] ?? regexes['en'];
-
 export class RegexesHolder {
-  StatsRegex: CactbotBaseRegExp<'PlayerStats'>;
-  YouGainEffectRegex: CactbotBaseRegExp<'GainsEffect'>;
-  YouLoseEffectRegex: CactbotBaseRegExp<'LosesEffect'>;
-  YouUseAbilityRegex: CactbotBaseRegExp<'Ability'>;
-  AnybodyAbilityRegex: CactbotBaseRegExp<'Ability'>;
-  MobGainsEffectRegex: CactbotBaseRegExp<'GainsEffect'>;
-  MobLosesEffectRegex: CactbotBaseRegExp<'LosesEffect'>;
-  MobGainsEffectFromYouRegex: CactbotBaseRegExp<'GainsEffect'>;
-  MobLosesEffectFromYouRegex: CactbotBaseRegExp<'LosesEffect'>;
-  cordialRegex: CactbotBaseRegExp<'Ability'>;
+  cordialRegex: RegExp;
   countdownStartRegex: RegExp;
   countdownCancelRegex: RegExp;
   craftingStartRegexes: RegExp[];
   craftingFinishRegexes: RegExp[];
   craftingStopRegexes: RegExp[];
 
-  constructor(lang: Lang, playerName: string) {
-    this.StatsRegex = NetRegexes.statChange();
-
-    this.YouGainEffectRegex = NetRegexes.gainsEffect({ target: playerName });
-    this.YouLoseEffectRegex = NetRegexes.losesEffect({ target: playerName });
-    this.YouUseAbilityRegex = NetRegexes.ability({ source: playerName });
-    this.AnybodyAbilityRegex = NetRegexes.ability();
-    this.MobGainsEffectRegex = NetRegexes.gainsEffect({ targetId: '4.{7}' });
-    this.MobLosesEffectRegex = NetRegexes.losesEffect({ targetId: '4.{7}' });
-    this.MobGainsEffectFromYouRegex = NetRegexes.gainsEffect({
-      targetId: '4.{7}',
-      source: playerName,
-    });
-    this.MobLosesEffectFromYouRegex = NetRegexes.losesEffect({
-      targetId: '4.{7}',
-      source: playerName,
-    });
+  constructor(lang: Lang) {
+    const localeRegexes = getLocaleRegexes(lang);
+
     // use of GP Potion
-    this.cordialRegex = NetRegexes.ability({
-      source: playerName,
-      id: '20(017FD|F5A3D|F844F|0420F|0317D)',
-    });
-
-    const getCurrentRegex = getLocaleRegex.bind(this, lang);
-    this.countdownStartRegex = getCurrentRegex(LocaleNetRegex.countdownStart);
-    this.countdownCancelRegex = getCurrentRegex(LocaleNetRegex.countdownCancel);
+    this.cordialRegex = /20(017FD|F5A3D|F844F|0420F|0317D)/;
+
+    this.countdownStartRegex = localeRegexes.countdownStart;
+    this.countdownCancelRegex = localeRegexes.countdownCancel;
     this.craftingStartRegexes = [
-      LocaleNetRegex.craftingStart,
-      LocaleNetRegex.trialCraftingStart,
-    ].map(getCurrentRegex);
+      localeRegexes.craftingStart,
```

Comment:
The if/then => switch changes look fine, but I'm not sure I understand what the goal here is.  It seems like you could have left the original `getLocaleRegex` code and your "matching via logDefinition" purpose would still be met?

I think if you don't like the ordering of line => lang => regex and would rather index on lang => line => regex, I'd rather change `LocaleRegexesObj` to do that rather than having `LocaleRegexesObj` and `getLocaleRegexes` as two separate ways of building the same information, if that makes sense.  (And I think that would make more sense as a separate change than to put in here, unless I'm misunderstanding.)

Reply:
Yeah, you are right. But I think it looks like it is redundant to match/extract something like `timeline` or `type` that we don't need, and that I think it should improve performance a bit? Because extracting items from an array is just `O(1)`.

I wanted to replace the whole `RegexSet` in `translation.ts` to `getLocaleRegexes` mentioned above, but there are not only `jobs` module relies on them, so I leave them still alive as this PR changed `jobs` module only. So it is not an ordering thing? I am just changing the regex match range/scope to the message only....

Reply:
Sorry, by `timeline` and `type`, what are you referring to?

And yeah, other things use `LocalRegexesObj` / `RegexSet`, so those would need to be changed as well.  That's why I was suggesting to do that in another PR where you could change all of them at once to something more convenient.

Reply:
Oh, sorry, it is `timestamp` and `code`....
I am referring that when we are using the RegExp that generated from NetRegexes, the full regex would be like (in jp):
```js
/^(?<type>(?:00))\|(?<timestamp>(?:[^|]*))\|(?<code>(?:[^|]*))\|(?<name>(?:[^|]*))\|(?<line>(?:戦闘開始まで(?<time>-?[0-9]+(?:[.,][0-9]+)?(?:E-?[0-9]+)?)秒！ （(?<player>.*?)）[^|]*?))\|/
```
There are many match groups, but we need only is the `line` group, others like `timestamp` or `code` are not required in this situation, right? I think as we have the `logDefinition` that records all items' positions already, we can directly extract the `line` part via an array indexing (`log[fields.line]`), instead to do a longer regex matching.

Reply:
Sorry for any misunderstanding, but I think we are talking about different things.

I am trying to say that I think your patch is doing two different (unrelated) things:
(1) remove a bunch of regexes in favor of a proxy / `logDefinitions`, change `if` to `switch (e.g. `YouGainEffectRegex`)
(2) change the initialization of other regexes that you haven't removed (e.g. `countdownStartRegex`)

I am saying that (1) should be this PR on its own and is all good.  I don't have any more feedback on that.  I think what you are responding to above in the previous comment is about (1), if I understand correctly.  Is it possible to make this PR only do (1) and not (2), and then I would merge it?

The (2) part is what I was trying to get at with these comments.  I was mostly saying that it feels like you are adding a second method to build/access `LocaleRegexesObj` / `RegexSet` and I think having both `buildLocaleRegexes` and `getLocaleRegexes` is somewhat redundant.  If you want to change how `translations.ts` works, then I think it'd make sense to do that in a separate PR and change other parts of the code that use it.

Reply:
Ah, I see your point. Sorry for the unrelated responses.
Sure, I can try to separate the (2) things into a new PR later.